ray_provider/operators/ray.py

```diff
@@ -350,14 +315,20 @@ def execute_complete(self, context: Context, event: dict[str, Any]) -> None:
         """
         Handle the completion of a deferred Ray job execution.
 
+        This method is called when the deferred job execution completes. It processes the final
+        job status and raises exceptions for failed or cancelled jobs. It finally deletes the cluster when the ray spec is provided
+
         :param context: The context in which the task is being executed.
         :param event: The event containing the job execution result.
-        :raises AirflowException: If the job execution fails or is cancelled.
+        :raises AirflowException: If the job execution fails, is cancelled, or reaches an unexpected state.
         """
-        if event["status"] in [JobStatus.STOPPED, JobStatus.FAILED]:
-            self.log.info(f"Ray job {self.job_id} execution not completed...")
-            raise AirflowException(event["message"])
-        elif event["status"] == JobStatus.SUCCEEDED:
-            self.log.info(f"Ray job {self.job_id} execution succeeded ...")
-        else:
-            raise AirflowException(f"Unexpected event status: {event['status']}")
+        try:
+            if event["status"] in [JobStatus.STOPPED, JobStatus.FAILED]:
+                self.log.info(f"Ray job {self.job_id} execution not completed successfully...")
+                raise AirflowException(f"Job {self.job_id} {event['status'].lower()}: {event['message']}")
+            elif event["status"] == JobStatus.SUCCEEDED:
+                self.log.info(f"Ray job {self.job_id} execution succeeded.")
+            else:
+                raise AirflowException(f"Unexpected event status for job {self.job_id}: {event['status']}")
+        finally:
+            self._delete_cluster()
```

Comment:
I'll refer to it as behavior change rather than enhancement, so we should consider it a breaking change.

Reply:
Also, I think we should add a parameter to control this and only delete cluster for the user who want to delete it

Reply:
Parameter is already added "if self.ray_cluster_yaml:"..

If the cluster spec is not provided, then the SubmitRayJob operator behavior does not change from the previous release. That is also why this is an enhancement and not a breaking change.